frontend/lib/src/components/graph.dart

```diff
@@ -19,138 +18,213 @@ class Graph extends StatelessWidget {
+                BoxShadow(
+                  color: Color.fromRGBO(0, 0, 0, 0.25),
+                  blurRadius: 2,
+                  spreadRadius: 1,
+                  offset: Offset(0, 2),
+                ),
+              ],
+            ),
+            padding: const EdgeInsets.all(20),
+            child: InkWell(
+              onTap: onTap,
+              child: SizedBox(
+                width: double.maxFinite,
+                height: 236,
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    SizedBox(
+                      width: 150,
+                      height: 150,
+                      child: StatefulBuilder(
+                        builder: (context, stateful) {
+                          return Obx(
+                            () => PieChart(
+                              PieChartData(
+                                pieTouchData: PieTouchData(
+                                  touchCallback:
+                                      (FlTouchEvent event, pieTouchResponse) {
+                                    stateful(() {
+                                      if (!event.isInterestedForInteractions ||
+                                          pieTouchResponse == null ||
+                                          pieTouchResponse.touchedSection ==
+                                              null) {
+                                        graphController.setFocusIndex = -1;
+
+                                        return;
+                                      }
+                                      graphController.setFocusIndex =
+                                          pieTouchResponse.touchedSection!
+                                              .touchedSectionIndex;
+                                    });
+                                  },
+                                ),
+                                borderData: FlBorderData(
+                                  show: false,
+                                ),
+                                sectionsSpace: 0,
+                                centerSpaceRadius: 30,
+                                sections: showingSections(),
+                              ),
+                            ),
+                          );
+                        },
+                      ),
                     ),
-                    const SizedBox(
-                      height: 18,
+                    Column(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: legendSection(legendList),
                     ),
                   ],
                 ),
-              ],
+              ),
             ),
           ),
-        ),
-      ),
+        );
+      },
     );
   }
 
+  List<Widget> legendSection(List<TransactionCategory> legendList) {
```

Comment:
use Indicator widget type

Reply:
nvm this, since youre comboing it with the sizedbox

Reply:
Will this work if i have space between each indicator?

![image](https://github.com/framgia/sph-flutter/assets/114897466/5143eda1-7edb-4f9b-b7f1-77919d457dc1)

Reply:
yeah i realized it late